### DISCUSSION TITLE
Simplify(filter_nodes=False)

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/what-is-an-arg-paper#92] [Q&A]
We've never had use for this before, but since we are using the simplify() command to turn an implicit ARG into an explicit one, there's a good argument that we should be able to run the command but *keep the non-ancestral nodes*. One way to do this is to have an additional parameter "filter_nodes" (default False) to `simplify`.

**Comment by jeromekelleher** [comment]
Sure - not clear how much work this would be to do, though, and whether it would have much application outside of thinking about Big ARGs and such.

**Reply by hyanwong** [reply]
Yes, I wasn't sure whether this was generally useful or not. One nice thing is that it doesn't change the node IDs though. Here's a (poorly tested) function to do it for us, anyway. I could add this to the argutils/ancestry.py file?

```python
def simplify_edges_only(ts):
    """
    Modify the edges using simplify, but leave other tables intact
    """
    # get the edges to keep
    ts2, node_map = ts.simplify(keep_unary=True, map_nodes=True)
    val, inverted_map = np.unique(node_map, return_index=True)
    inverted_map = inverted_map[val != tskit.NULL]
    # only use the edges in the simplified one, but keep all other tables, including nodes etc, from the original
    tables = ts.dump_tables()
    tables.edges.clear()
    for edge in ts2.tables.edges:
        tables.edges.append(edge.replace(child=inverted_map[edge.child], parent=inverted_map[edge.parent]))
    return tables.tree_sequence()
```
